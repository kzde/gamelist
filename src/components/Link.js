import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { themeColors } from '../utils/styleUtils';

const Wrapper = styled.div`
  display: inline-block;
  outline: none;
  overflow: hidden;
  position: relative;
  cursor: pointer;

  &:hover span,
  &:active span {
    transform: translateY(-100%);
    color: ${themeColors.link};
  }
`;
const Content = styled.span`
  display: inline-block;
  position: relative;
  transition: transform 0.3s;
  color: ${props => props.textColor};

  &::before {
    position: absolute;
    content: attr(data-hover);
    top: 100%;
    transform: translate3d(0, 0, 0);
  }
`;

const Link = props => (
  <Wrapper onClick={() => props.handleClick()}>
    <Content data-hover={props.text} textColor={props.textColor}>
      {props.text}
    </Content>
  </Wrapper>
);

Link.propTypes = {
  handleClick: PropTypes.func,
  text: PropTypes.string.isRequired,
  textColor: PropTypes.string
};
Link.defaultPropTypes = {
  handleClick: () => {},
  textColor: 'black'
};
export default Link;
