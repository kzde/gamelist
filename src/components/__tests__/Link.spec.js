import React from 'react';
import renderer from 'react-test-renderer';
import Link from '../Link';

describe('Link', () => {
  it('should renders correctly', () => {
    const tree = renderer.create(<Link text="link test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
