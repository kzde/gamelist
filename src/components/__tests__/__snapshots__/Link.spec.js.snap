// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should renders correctly 1`] = `
<div
  className="Link__Wrapper-sc-1kz7l5q-0 jgxdgC"
  onClick={[Function]}
>
  <span
    className="Link__Content-sc-1kz7l5q-1 gOEnav"
    data-hover="link test"
  >
    link test
  </span>
</div>
`;
